hal/armv7m/imxrt/10xx/hal.c

```diff
@@ -43,18 +43,13 @@ extern void console_init(void);
 void hal_init(void)
 {
 	_imxrt_init();
+	_imxrt_scbSetPriorityGrouping(3);
+
 	mpu_init();
 	timer_init();
-
 	console_init();
 
-	syspage_init();
-	syspage_setAddress(ADDR_SYSPAGE);
-
-	_imxrt_scbSetPriorityGrouping(3);
-
-	/* Add entries related to plo image */
-	syspage_addEntries((addr_t)_plo_bss, (addr_t)_end - (addr_t)_plo_bss + SIZE_STACK);
+	hal_common.halSyspage = (syspage_hal_t *)ADDR_SYSPAGE;
```

Comment:
I suggest to add special function for providing address of the syspage into the HAL.